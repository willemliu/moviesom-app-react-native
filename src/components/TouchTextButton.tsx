import React from 'react';
import { Platform, Text, GestureResponderEvent, TextStyle, StyleProp, View } from 'react-native';
import { textStyle, touchTextButtonStyle } from '../styles/Styles';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import Touchable from './Touchable';

export interface Props {
    onPress: (event: GestureResponderEvent) => void;
    style?: StyleProp<TextStyle>;
    children?: any;
    textStyle?: StyleProp<TextStyle>;
}

export default class TouchTextButton extends React.Component<Props, any> {
    render() {
        return (
            <Touchable onPress={this.props.onPress}>
                <View style={[touchTextButtonStyle.view, this.props.style]}>
                    <Text style={[touchTextButtonStyle.text, this.props.textStyle]}>{this.props.children}</Text>
                </View>
            </Touchable>
        );
    }
}
