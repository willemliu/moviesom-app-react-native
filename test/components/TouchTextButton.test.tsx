import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

import React from 'react';
import renderer from 'react-test-renderer';
import TouchTextButton from '../../src/components/TouchTextButton';

Enzyme.configure({ adapter: new Adapter() });

describe('TouchTextButton', () => {
    it('should render the correct TouchTextButton component', () => {
        const spy = jest.fn();
        const rendered = renderer.create(<TouchTextButton onPress={spy}>Touch text button</TouchTextButton>).toJSON();
        expect(rendered).toMatchSnapshot();
    });

    it('should handle onPress event', () => {
        const spy = jest.fn();
        const rendered = shallow(<TouchTextButton onPress={spy}>Touch text button</TouchTextButton>);
        expect(spy).toHaveBeenCalledTimes(0);
        rendered.simulate('press');
        expect(spy).toHaveBeenCalled();
    });
});
