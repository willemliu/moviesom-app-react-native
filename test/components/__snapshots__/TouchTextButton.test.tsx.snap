// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchTextButton should render the correct TouchTextButton component 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeBackgroundAndroid={
    Object {
      "attribute": "selectableItemBackground",
      "type": "ThemeAttrAndroid",
    }
  }
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#008CBA",
        "borderColor": "#008CBA",
        "borderRadius": 3,
        "borderWidth": 1,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Touch text button
  </Text>
</View>
`;
